### ISSUE TITLE
ColorField to get de-focused on submit (post enter key press)

### ISSUE THREAD

**Issue by rohijai** [adobe/react-spectrum#2596]
# 🙋 Feature Request
When user is done modifying value in ColorField and enter is pressed to submit it then highlighting (blue border) of ColorFiled should be turned off but keyboard focus should be on it only, i.e. when TAB is pressed then focus should move to next element.

## 🤔 Expected Behavior
Blue color border should be turned off when enter is pressed i.e. onSubmit

## 😯 Current Behavior
Blue color border is not turned off when enter is pressed i.e. onSubmit

## 🔦 Context
As part of Adobe Color we are providing Color-Picker component which is integrated in CC Web Canvas (https://stage.creativecloud.adobe.com/cc/canvas), for all elements as on canvas this behaviour can be observed.

We want to provide consistency in behaviour. 

## 💻 Examples

**Expected SS**
<img width="2005" alt="Expected" src="https://user-images.githubusercontent.com/89633686/143176894-dd002a30-f75a-4c10-97a6-145583198b79.png">

**Current SS**
<img width="882" alt="Current" src="https://user-images.githubusercontent.com/89633686/143176912-82b0024a-72f4-4854-8bc9-bbc57db91e26.png">

## 🧢 Your Company/Team
Adobe Color

**Comment by snowystinger** [comment]
I believe your example is an accessibility violation. Focus must be visible at all times for keyboard users. See https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

**Comment by rohijai** [comment]
Hey. This has been discussed with Adobe Color PM/design and they want it fixed.

**Comment by majornista** [comment]
> Hey. This has been discussed with Adobe Color PM/design and they want this feature.

We should not implement a behavior that violates a WCAG accessibility requirement. When the user is interacting using the keyboard, the focus ring must be visible. It sounds like design is looking for some visual confirmation that a the value has been submitted. ColorField and NumberField support [`validationState`](https://react-spectrum.adobe.com/react-spectrum/ColorField.html#validation), could we use that to confirm valid input, without removing the focus ring from a field that still has focus.
